app/models/profile.rb

```diff
@@ -62,9 +62,34 @@ def encrypt_recovery_pii(pii)
     @personal_key = personal_key
   end
 
+  def self.build_compound_pii_fingerprint(pii)
+    return unless FeatureManagement.enable_compound_pii_fingerprint?
+
+    values = [
+      pii.first_name,
+      pii.last_name,
+      pii.zipcode,
+      pii.dob && Date.parse(pii[:dob]).year,
+    ]
+
+    return unless values.all?(&:present?)
+
+    Pii::Fingerprinter.fingerprint(values.join(':'))
+  end
+
   private
 
   def personal_key_generator
     @_personal_key_generator ||= PersonalKeyGenerator.new(user)
   end
+
+  def encrypt_ssn_fingerprint(pii)
+    ssn = pii.ssn
+    self.ssn_signature = Pii::Fingerprinter.fingerprint(ssn) if ssn
+  end
+
+  def encrypt_compound_pii_fingerprint(pii)
+    compound_pii_fingerprint = self.class.build_compound_pii_fingerprint(pii)
+    self.name_zip_birth_year_signature = compound_pii_fingerprint if compound_pii_fingerprint
```

Comment:
I didn't like `pii_fingerprint` since that made it sound it encompasses all the PII rather than just a few key attributes

So I named it for what it contains (in case we want to add a fingerprint on different attributes in the future?)